site/en/blog/blinkng/index.md

```diff
@@ -0,0 +1,185 @@
+
+## Benefits
+
+As you've seen, a well-defined rendering pipeline yields enormous long-term benefits. There are even more than you might think:
+
+-  **Greatly improved reliability**: This one is pretty straightforward. Cleaner code with well-defined and understandable interfaces is easier to understand, write, and test. This makes it more reliable. It also makes the code safer and more stable, with fewer crashes and fewer use-after-free bugs.
+-  **Expanded test coverage**: In the course of Rendering NG, we added a great many new tests to our suite. This includes unit tests that provide focused verification of internals; regression tests that prevent us from reintroducing old bugs we've fixed (so many!); and lots of additions to the public, collectively-maintained [Web Platform Test suite](https://wpt.fyi), which all browsers use to measure conformance to web standards.
+-  **Easier to extend**: If a system is broken down into clear components, it's not necessary to understand other components at any level of detail in order to make progress on the current one. This makes it easier for everyone to add value to the rendering code without having to be a deep expert, and it also makes it easier to reason about the behavior of the entire system.
+-  **Performance**: Optimizing algorithms written in spaghetti code is difficult enough, but it's almost impossible to achieve even bigger things such as [universal threaded scrolling and animations](/blog/renderingng/#threaded-scrolling-animations-and-decode) or the [processes and threads for site isolation](/blog/renderingng-architecture/#process-and-thread-structure) without such a pipeline. Parallelism can help us to improve performance tremendously, but is also extremely complicated.
+-  **Yielding and containment**: There are several new features made possible by RenderingNG that exercise the pipeline in new and novel ways. For example, what if we wanted to only run the rendering pipeline until a budget has expired? Or skip rendering for subtrees known to be not relevant to the user right now? That's what the [content-visibility](/blog/renderingng/#extensibility-the-right-tools-for-the-job) CSS property enables. What about making the style of a component depend on its layout? That's [container queries](/blog/renderingng/#extensibility-the-right-tools-for-the-job).
+
+## Case study: Container queries
+
+Container queries is a highly anticipated upcoming web platform feature (it has been the number one most requested feature from CSS developers for years). If it's so great, why doesn't it exist yet? The reason is that an implementation of container queries requires very careful understanding and control of the relationship between the style and layout code. Let's take a closer look.
```

Comment:
"container queries requires containment" should be "container queries require containment".

Reply:
I couldn't find that phrase, "The reason is that an implementation of container queries requires very careful understanding and control of the relationship between the style and layout code." seems right.

Reply:
You're right, my mistake.